src/components/FilterBar/FilterDisplay/Inputs/DateInput.react.js

```diff
@@ -17,6 +17,16 @@ export class DateInput extends React.Component {
     this.setState({value: newValue});
   }
 
+  onRelativeChange(event) {
+    var optionElement = event.target.childNodes[event.target.selectedIndex];
+    var from = moment(parseInt(optionElement.getAttribute('data-from')));
+    var to = moment(parseInt(optionElement.getAttribute('data-to')));
+    var newValue = { value: optionElement.value, from: from.format('DD/MM/YYYY'), to: to.format('DD/MM/YYYY') };
+
+    this.setState({value: newValue });
+    this.context.filterBarActor.updateFilter(this.props.filterUid, "value", newValue);
```

Comment:
I wanted to do the same as onBlur(), but this.state is not updated this time...?

Reply:
[Currently, `setState` is asynchronous inside event handlers](https://reactjs.org/docs/faq-state.html#when-is-setstate-asynchronous)

Reply:
@chesterl We have a few options here. We can use a callback on `setState`, the lifecycle event `componentDidUpdate`, or what you've done here. I think what you've got here works.